fdbserver/RestoreRoleCommon.actor.cpp

```diff
@@ -58,7 +58,7 @@ ACTOR Future<Void> handleFinishRestoreRequest(RestoreVersionBatchRequest req, Re
 	return Void();
 }
 
-ACTOR Future<Void> handleInitVersionBatchRequest(RestoreVersionBatchRequest req, Reference<RestoreRoleData> self) {
+Future<Void> handleInitVersionBatchRequest(RestoreVersionBatchRequest req, Reference<RestoreRoleData> self) {
```

Comment:
Do you know if there's a reason these are returning `Future`? If not, an alternate fix that may be more clear is to change the return type to `void` (and update the caller appropriately).

Reply:
They are added into actor collections and thus must return a `Future`.

Reply:
Do they need to be in the actor collection?

Reply:
I guess @xumengpanda has a better explanation but based on my understanding, the fast restore code is still under developed and thus, although the actor collection is not currently used anywhere which means we can remove those functions out of it and therefore simply return `Void()`, I don't know what's the plan for that.

Reply:
I think one thing that the actor collection would do differently is die immediately if any of the inserted futures is an error without failing the calling actor. I'm not sure if that's the desired behavior or not (or maybe it's moot if none of these things throw errors).

We should check with @xumengpanda, and if he doesn't have any reason for keeping them like this I would be in favor of simplifying them to not return futures.

Reply:
I think returning void is better. 
The caller should not proceed before processing this request.

Reply:
On it.

Reply:
I can fix this in my branch, BTW. I'm also happy if you want to revise this PR.

Reply:
Oh great I will close this PR and leave this to you to clean up to avoid rebasing pains on your side.